EventSubscriber/NavigationInvalidationSubscriber.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\EventSubscriber;
+
+use PHPCR\SessionInterface;
+use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentInspector;
+use Sulu\Bundle\DocumentManagerBundle\Bridge\PropertyEncoder;
+use Sulu\Bundle\HttpCacheBundle\Cache\CacheManager;
+use Sulu\Component\DocumentManager\Event\PublishEvent;
+use Sulu\Component\DocumentManager\Event\RemoveEvent;
+use Sulu\Component\DocumentManager\Event\RemoveLocaleEvent;
+use Sulu\Component\DocumentManager\Event\UnpublishEvent;
+use Sulu\Component\DocumentManager\Events;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class NavigationInvalidationSubscriber implements EventSubscriberInterface
```

Comment:
```suggestion
/**
 * @final
 * @internal
 */
class NavigationInvalidationSubscriber implements EventSubscriberInterface
```

Should be kept internal and final.